pkg/cli/debug/debug.go

```diff
@@ -1063,7 +1061,19 @@ func (o *DebugOptions) approximatePodTemplateForObject(object runtime.Object) (*
 					},
 				},
 			},
-		}, nil
+		}
+		if t.Labels[corev1.LabelOSStable] == "windows" {
+			template.Spec.OS = &corev1.PodOS{Name: corev1.Windows}
+			template.Spec.HostPID = false
```

Comment:
We are setting HostPID and HostIPC false, I wonder how user will debug node from this pod? Because there seems to be no privilege to access the node components.

Reply:
HostPID and HostIPC are not usable with Windows pods:
https://kubernetes.io/docs/concepts/windows/intro/#compatibility-linux-similarities

```
...
If you set the .spec.os.name field to windows, you must not set the following fields in the .spec of that Pod:

    spec.hostPID
    spec.hostIPC
...
```

Reply:
In addition to your comment, according to here https://github.com/openshift/oc/blob/4eb0aab1f3183374b543fab7ef04e29b56b560c5/vendor/k8s.io/api/core/v1/types.go#L3426-L3450, these fields should be cleaned to default values which is false.